below/render/src/lib.rs

```diff
@@ -89,6 +91,77 @@ pub struct RenderConfigBuilder {
     rc: RenderConfig,
 }
 
+#[derive(Clone)]
+pub enum OpenMetricsType {
+    /// Gauges are current measurements, such as bytes of memory currently used or the number of
+    /// items in a queue. For gauges the absolute value is what is of interest to a user.
+    Counter,
+    /// Counters measure discrete events. Common examples are the number of HTTP requests received,
+    /// CPU seconds spent, or bytes sent. For counters how quickly they are increasing over time is
+    /// what is of interest to a user.
+    Gauge,
+}
+
+/// Configuration for rendering fields in OpenMetrics format.
+///
+/// See the OpenMetrics spec for more details:
+/// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md
+#[derive(Clone)]
+pub struct RenderOpenMetricsConfig {
+    ty: OpenMetricsType,
+    help: Option<String>,
+}
+
+#[derive(Clone)]
+pub struct RenderOpenMetricsConfigBuilder {
```

Comment:
Haven't heard of it before, but looks neat. While I think it could work, we'd still need to do custom impls on the generated builder for `label()` and `new()`. It also looks like the proc macro always wants to generate a trail impl for `Default` which isn't really what we want here b/c we want to require the metric type always be present. Right now it's a compile time error not to provide metric type. But if default() was used and metric type is not present, it's a runtime error.